java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -80,7 +89,34 @@ private static boolean isExcluded(MethodTree tree) {
   }
 
   private static boolean isAnnotated(MethodTree tree) {
-    return !tree.modifiers().annotations().isEmpty();
+    if (tree.modifiers().annotations().isEmpty()) {
+      return false;
+    }
+
+    // If any annotation doesn't match the @SuppressWarning then mark the method as annotated.
+    return tree.modifiers().annotations().stream().anyMatch(annotation -> !isExcludedSuppressWarningAnnotation(annotation));
+  }
+
+  private static boolean isExcludedSuppressWarningAnnotation(AnnotationTree tree) {
+    if (!tree.annotationType().symbolType().is(SUPPRESS_WARNINGS_ANNOTATION)) {
+      return false;
+    }
+
+    return tree.arguments().stream().allMatch(UnusedMethodParameterCheck::containsOnlyExcludedWarnings);
+  }
+
+  private static boolean containsOnlyExcludedWarnings(Tree tree) {
+    if (tree.is(Tree.Kind.STRING_LITERAL)) {
```

Comment:
I tend to think that this should be written as a single method (`isExcludedLiteral` maybe ?) that could be used as a method ref in the unwrapping of the newArrayTree and would avoid you write `return false`  but `return isExcludedLiteral(tree);` (bit more compact)